### PR TITLE
Support up to 9.8

### PR DESCRIPTION
**Pull request by tomjaguarpaw**

### CODE REVIEW

**Comment by tomjaguarpaw** [comment]
Actually, only up to 9.6, because `hedgehog` has not caught up with 9.8 yet.

File: extensions.cabal [hunk 1]
```diff
@@ -80,14 +82,23 @@ library
...
+                       -- We need to pin a single major version of
+                       -- Cabal here because the main reason we use
+                       -- Cabal is for its list of extensions.  Later
+                       -- versions have strictly more extensions, and
+                       -- we'll have missing patterns if we try to
+                       -- support more than one major version.  If
+                       -- this causes problems in practice let's
+                       -- revisit this decision and come up with
+                       -- another approach.
+                     , Cabal ^>= 3.10
```
Comment (ulidtko) [inline extensions.cabal:95]: Spinned off a discussion in #111.

This hit me when trying to package `extensions` @ the latest for ArchLinux which is still on GHC 9.4.8 (unfortunately). I had to pin the package to `0.1.0.0`.

Comment (tomjaguarpaw) [reply extensions.cabal:95]: Thanks, is there anything I need to do to support you, or is this just for information, like https://github.com/kowainik/extensions/issues/111?

Comment (ulidtko) [reply extensions.cabal:95]: Just informational, thanks :pray:

**Pull request merged**